peterpy/services/product_service.py

```diff
@@ -5,67 +5,53 @@
 from peterpy.entities import Product
 from peterpy.interfaces import IRepository
 
-from peterpy.services.kafka_service import KafkaService
-
 
 class ProductService:
     def __init__(self, repository: IRepository[Product]):
-        self._repository = repository
-        self.kafka_service = KafkaService("products")
-
-    async def add(self, name: str, price: float) -> Product:
-        try:
-            if len(self._repository.find({"name": name})) > 0:
-                raise ValueError(f"Product with name {name} already exists")
-
-            product = Product(name=name, price=price)
-            logging.debug(f"Adding product {product}")
-            logging.debug(product.__dict__)
-            self._repository.add(product)
-
-            await self.kafka_service.produce(
-                {"event_name": "ProductAdded", "product": product.to_json()}
-            )
-
-            return product
-        except Exception as err:
-            logging.error("Error adding product: %s", err)
-            raise
-
-    def update(self, id, name, price):
-        product = self._repository.get(id)
+        self.repository = repository
+
+    async def add(self, name: str, price: float):
+        if len(self.repository.find({"name": name})) > 0:
```

Comment:
I would let the database deal with duplicates instead of on code level. Now you are adding a performance hit for something that shouldn't happen that often